### PR TITLE
Document and IO Type Fixes

### PR DESCRIPTION
**Pull request by VikasKumarC**
Signed-off-by: Vikas Kumar Choudhary <vikas.kumar@seagate.com>

## Problem Statement

-   Document linting issues  and Code Fixes

## Design

-   For Bug, Describe the fix here.
-   For new test addition post the link for design

## Coding

   Checklist for Author

-   [x]  Coding conventions are followed and code is formatted (e.g. using pycharm in-built formatter)

## Testing

  Checklist for Author

-   [ ]  New/Affected tests are executed
-   [ ]  Attach test execution logs

## Review 

  Checklist for Author

-   [x]  JIRA number/GitHub Issue added to PR
-   [ ]  PR is self reviewed
-   [ ]  Jira and state/status is updated and JIRA is updated with PR link
-   [x]  Check if the description is clear and explained

## Documentation

  Checklist for Author

-   [x]  Changes done to ReadMe

-----
[View rendered docs/Architecture_and_Design.md](https://github.com/VikasKumarC/corio/blob/cortx-30427-Doc_IOType-Fixes/docs/Architecture_and_Design.md)
[View rendered docs/YAML_documents/yaml_structure.md](https://github.com/VikasKumarC/corio/blob/cortx-30427-Doc_IOType-Fixes/docs/YAML_documents/yaml_structure.md)

### CODE REVIEW

File: docs/YAML_documents/yaml_structure.md [hunk 1]
```diff
@@ -41,6 +47,19 @@ as total number of sessions for a test workload. Default value of number_of_node
...
 
+**write_percentage** is percentage of data to fill the storage size.
+
+**read_percentage** is percentage of data to read from given storage size.
+
+**delete_percentage** is percentage of data to be deleted from storage.
+
+**cleanup_percentage** is Percentage(less or equal to write) of data once reached then do cleanup.
+
+**total_storage_size** Total available storage from cluster or user inout size will be used as following
```
Comment (priyankaborawake) [inline docs/YAML_documents/yaml_structure.md]: entered*

File: workload/s3/mix_io/type3_mix_crud_object_operations.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ test_1:
     write_percentage: 100  # Percentage of data to fill the storage.
     read_percentage: 100  # Percentage of data to be used to read from storage.
     delete_percentage: 0  # Percentage of data to be deleted from storage.
-    cleanup_percentage: 100  # Percentage of data once reached then cleanup all data.
+    cleanup_percentage: 0  # Percentage of data once reached then cleanup all data.
```
Comment (priyankaborawake) [inline workload/s3/mix_io/type3_mix_crud_object_operations.yaml:9]: cleanup should be around 95percent

**Review by priyankaborawake — CHANGES_REQUESTED** [review]
Requested few minor changes.

File: docs/Architecture_and_Design.md [hunk 3]
```diff
@@ -1,43 +1,42 @@
...
-- If test stopped before given time due to any issue:
-    - Report issue and Stop for debugging
-- Check the Stop Conditions
-- Mark test pass/fail (in JiraTest Plan) 
-- Collect Support Bundle irrespective of test status continuously
-- If Test fails, it collects Support Bundle logs and upload to designated LOCAL/NFS (from server machine)
-- Arrange/Collect and push Test/Tools o/p IO logs to designated LOCAL/NFS Server (from Client machine)
-- Start next test if previous tests run successfully for defined minimum run time
+* Start multiple workload configuration(tests) in parallel:  
+  * Record the test thread/process for status monitoring
```
Comment (DhobaleVishal) [inline docs/Architecture_and_Design.md:31]: Please check if indentation is correct with this changes.

File: src/libs/s3api/s3_object_ops.py [hunk 4]
```diff
@@ -273,7 +273,7 @@ def delete_s3_objects(self, bucket_name, object_prefix=None):
             objects = []
             for file in self.get_boto3_resource().Bucket(bucket_name).objects.all():
                 if file.key.startswith(object_prefix):
-                    objects.append({"key": file.key})
```
Comment (DhobaleVishal) [inline src/libs/s3api/s3_object_ops.py]: Please, take latest pull as issue fixed and verified in https://github.com/Seagate/corio/pull/66

**Review by DhobaleVishal — CHANGES_REQUESTED** [review]

File: docs/Architecture_and_Design.md (hunk 3 above)
Comment (VikasKumarC) [reply docs/Architecture_and_Design.md:31]: Done

File: docs/YAML_documents/yaml_structure.md (hunk 1 above)
Comment (VikasKumarC) [reply docs/YAML_documents/yaml_structure.md]: Done

File: src/libs/s3api/s3_object_ops.py (hunk 4 above)
Comment (VikasKumarC) [reply src/libs/s3api/s3_object_ops.py]: ok

**Review by DhobaleVishal — APPROVED** [review]

**Review by priyankaborawake — APPROVED** [review]

**Pull request merged**